pkg/controllers/workgenerator/controller.go

```diff
@@ -315,10 +384,10 @@ func (r *Reconciler) syncAllWork(ctx context.Context, resourceBinding *fleetv1be
 		newWork = append(newWork, work)
 
 		// issue all the create/update requests for the corresponding works for each snapshot in parallel
-		for i := range newWork {
-			work := newWork[i]
+		for ni := range newWork {
+			w := newWork[ni]
```

Comment:
any particular reason for the local var name change?

Reply:
try to avoid the shadow. in case, we accidently use "=" instead of ":=" and the value will be overwritten.